custom_components/presence_simulation/__init__.py

```diff
@@ -264,7 +269,19 @@ async def restart_presence_simulation(call, entities_after_restart=None, delta_a
     async def simulate_single_entity(entity_id, hist, overridden_delta, overridden_random):
         """This method will replay the historic of one entity received in parameter"""
         _LOGGER.debug("Simulate one entity: %s", entity_id)
-        for state in hist: #hypothsis: states are ordered chronologically
+
+        # hist contains state changes starting HISTORY_EXTRA_DAYS days before the sim interval.
```

Comment:
I don't understand the reason of this. You use HISTORY_EXTRA_DAYS, but if you have a light that haven't been turned on or off during that additional day, it won't do what you expect it to do.
Furthermore, when I retrieve the historic, I call `get_significant_states()` giving it the parameter `include_start_time_state` to `true` (code [here](https://github.com/home-assistant/core/blob/57239769bae96ee4a49f4d0644890c613555c99a/homeassistant/components/recorder/history.py#L217) and [here](https://github.com/home-assistant/core/blob/57239769bae96ee4a49f4d0644890c613555c99a/homeassistant/components/recorder/history.py#L854)). With this parameter, the first state of the historic represents the state of the entity at the beginning of the period, which is exactly what we want, avoiding to retrieve a bigger period potentially insufficient

Reply:
Right, I removed it.